File: src/application/decorator/strategies/message_strategy.py
from application.decorator.strategies.abstracts.base_helper_strategy import BaseHelperStrategy
from application.decorator.strategies.decorator.decorator_helper_strategy import strategy
from application.enums.user_action import UserAction

@strategy(priority=999)  # sempre último
class MessageStrategy(BaseHelperStrategy):

    def can_handle(self, normalized: str, tokens: list[str]) -> bool:
        return True

    def handle(self, normalized: str, tokens: list[str]):
        return UserAction.MESSAGE, " ".join(tokens)

File: angular-app/src/core/application/decorator/helpers/strategies/message-strategy.ts
import { UserAction } from '../../../enums/user-action';
import { BaseHelperStrategy } from './abstracts/base-helper-strategy';

export class MessageStrategy extends BaseHelperStrategy {
  override priority = 999;

  override canHandle(normalized: string, tokens: string[]): boolean {
    return true;
  }

  override handle(normalized: string, tokens: string[]): [UserAction, string] {
    return [UserAction.MESSAGE, tokens.join(' ')];
  }
}
